### DISCUSSION TITLE
Backup anything (DB, Environment Variables etc..) with encryption

### DISCUSSION THREAD

**Discussion by luckycreationsindia** [coollabsio/coolify#2144] [Feature Requests]
Hi,

Can we add a feature to create a backup for DB, Environment Variables, Persistence Data or anything which I might be missing with encryption. We can ask user for password or security key or anything similar for additional security.

**Comment by swissbyte** [comment]
I would also like to see this.

Proposal:

Use restic

Minimum setup:

Provide another tab called backup, where the user cann add the restic backup configuration as bash script for example. Coolify should only take care about the regular execution of the script...

Basically, a "scheduled actions"
Per coolify, per project and per service would be nice. Just one shell script editor and a cron scheduling action.

Premium setup:
Use restic
Coolify takes care about the data to backup. Like automatically store mounts (* wirh the ability of disabling them) and also stores configuration alongside. Choose backup interval. Coolify should take care about the consistency of the backup with shuting down the container and starting up again. (Or use lvm for snapshots... but this needs to be supported by the host)

Features: 
- define backends for restic
- restore services directly from restic backend within coolify (generic backup structure with info about the service in a json)
- single file restore browser within coolify (restic has such a feature builtin)

That would put coolify to a next level comparable to proxmox with its proxmox backup server!

I would be happy to help implementing this! 

Just as a fun sidenote: there is also a proxmox backup client which can be used for non proxmox backups to a proxmox backup server 😁
